app/src/main/java/com/example/tasky/agenda/presentation/agenda_detail/AgendaDetailViewModel.kt

```diff
@@ -314,4 +328,39 @@ class AgendaDetailViewModel(
     private fun validateEmail(email: String): Boolean {
         return patternValidator.matches(email.trim())
     }
+
+    private fun saveTask() {
+        viewModelScope.launch(default) {
+            val currentTimestamp = ZonedDateTime.now(ZoneId.of("UTC"))
+            val task = Task(
+                id = UUID.randomUUID().toString(),
+                title = _state.value.title,
+                description = _state.value.description,
+                time = _state.value.fromTime.toKotlinInstant(),
+                remindAt = _state.value.fromTime
+                    .apply(_state.value.selectedAgendaReminderInterval)
+                    .toKotlinInstant(),
+                updatedAt = currentTimestamp.toKotlinInstant(),
+                isDone = _state.value.detailsAsTask()?.isDone ?: false
+            )
+            val result = taskRepository.createTask(task)
+            when (result) {
+                is Result.Error -> {
```

Comment:
This is a nice place to apply this [comment](https://github.com/ArturGubala/Tasky/pull/20/files#r2377270640) also 😊